src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py

```diff
@@ -397,112 +396,112 @@ def get_virtual_network_validator(has_type_field=False, allow_none=False, allow_
                                   default_none=False):
     from msrestazure.tools import is_valid_resource_id, resource_id
 
-    def simple_validator(namespace):
+    def simple_validator(cmd, namespace):
         if namespace.virtual_network:
             # determine if vnet is name or ID
             is_id = is_valid_resource_id(namespace.virtual_network)
             if not is_id:
                 namespace.virtual_network = resource_id(
-                    subscription=get_subscription_id(namespace.cmd.cli_ctx),
+                    subscription=get_subscription_id(cmd.cli_ctx),
                     resource_group=namespace.resource_group_name,
                     namespace='Microsoft.Network',
                     type='virtualNetworks',
                     name=namespace.virtual_network)
 
-    def complex_validator_with_type(namespace):
+    def complex_validator_with_type(cmd, namespace):
         get_folded_parameter_validator(
             'virtual_network', 'Microsoft.Network/virtualNetworks', '--vnet',
-            allow_none=allow_none, allow_new=allow_new, default_none=default_none)(namespace)
+            allow_none=allow_none, allow_new=allow_new, default_none=default_none)(cmd, namespace)
 
     return complex_validator_with_type if has_type_field else simple_validator
 
 
 # COMMAND NAMESPACE VALIDATORS
 
-def process_ag_listener_create_namespace(namespace):  # pylint: disable=unused-argument
+def process_ag_listener_create_namespace(cmd, namespace):  # pylint: disable=unused-argument
     from msrestazure.tools import is_valid_resource_id
     if namespace.frontend_ip and not is_valid_resource_id(namespace.frontend_ip):
         namespace.frontend_ip = _generate_ag_subproperty_id(
-            namespace, 'frontendIpConfigurations', namespace.frontend_ip)
+            cmd.cli_ctx, namespace, 'frontendIpConfigurations', namespace.frontend_ip)
```

Comment:
Am I under the right assumption that a few app-gateway/lb commands are not working right now? If yes, I suggest either unit test validators or add scenario tests with SDK's create method mock'd out to be a non-op.

Reply:
They were working because they were relying on `namespace.cmd.cli_ctx`.  They are covered by scenario tests. I wanted to update the syntax across the board to use the recommended signature instead of that old syntax.